qtcodes/circuits/circ.py

```diff
@@ -213,13 +219,40 @@ def cx(
         self,
         control: Union[TopologicalQubit, int],
         target: Union[TopologicalQubit, int],
+        ctype: Optional[str] = None,
+        params: Optional[Dict[str, int]] = None
+
     ):
+        """
+        CNOT operator on control and target topological qubit
+
+        Args: 
+            control (Union[TopologicalQubit, int]):
+                Either already a TopologicalQubit or an int index in treg
+            target (Union[TopologicalQubit, int]):
+                Either already a TopologicalQubit or an int index in treg
+            ctype (Optional[str]):
+                Specifies the logical type of ancilla bit
+            params (Optional[Dict[str, int]]):
+                Specifies the parameters of the ancilla bit
+        """
+
+        #default ctype and params
+        if ctype is not None or params is not None:
+            if ctype is None or params is None:
+                raise ValueError("Please provide both a ctype and params or neither to use the control qubit ctype and params by default.")
+            ctypes = [ctype]
+            params_list = [params]
```

Comment:
With the above change, you can just use `add_tqubit` directly without needing to create any lists.